### PR TITLE
chore(auth): Exclude URL from auth check as it can differ between environments

### PR DESCRIPTION
**Pull request by ram-senth**
By using URL as part of the payload that is signed we are assuming that URL used by client will be same as the URL seen in the request payload on Seer end. This is not true as requests from Subscription processor has full UTL including server name while those from the UI do not.

This PR makes URL optional. This will be step 1 of the fix. We need to follow this up with two more changes:
* remove URL from all locations on the client end where we are signing Seer requests
* once we confirm that Seer is not getting requests with URL anymore, then remove the code that checks for URL + payload.

### CODE REVIEW

File: src/seer/json_api.py [hunk 1]
```diff
@@ -163,18 +173,28 @@ def compare_signature(url: str, body: bytes, signature: str, config: AppConfig =
...
-    )
-
     for key in secrets:
-        computed = hmac.new(key.encode(), signature_input, hashlib.sha256).hexdigest()
-        is_valid = hmac.compare_digest(computed.encode(), signature_data.encode())
-        if is_valid:
+        if nonce:
+            if is_valid(b"%s:%s" % (nonce.encode(), body), key, signature_data):
+                sentry_sdk.metrics.incr(
+                    key="rpc_signature_auth_success", value=1, tags={"with_url": False}
```
Comment (ram-senth) [inline src/seer/json_api.py:180]: Change the key and value to `{"additional_payload": "nonce"}`? Similarly change line 193 to `{"additional_payload": "url"}`?

Comment (ram-senth) [reply src/seer/json_api.py:180]: I can also make the change if you are ok with it.

File: src/seer/json_api.py [hunk 2]
```diff
@@ -88,7 +88,10 @@ def wrapper(config: AppConfig = injected) -> Any:
             # Optional for now during rollout, make this required after rollout.
             if auth_header.startswith("Rpcsignature "):
                 parts = auth_header.split()
-                if len(parts) != 2 or not compare_signature(request.url, raw_data, parts[1]):
+                sentry_sdk.metrics.incr(key="rpc_signature_auth_attempt", value=1)
```
Comment (trillville) [inline src/seer/json_api.py:91]: metrics API is being deprecated, so you would want to switch it out for a span + span metric (e.g. https://github.com/getsentry/seer/blob/f393fef55f0446a98e16e081239459998aef0ec4/src/seer/app.py#L108-L109)

you can also do this to reference the current span https://github.com/getsentry/seer/blob/f393fef55f0446a98e16e081239459998aef0ec4/src/seer/grouping/grouping.py#L312

**Review by trillville — APPROVED** [review]

File: src/seer/json_api.py (hunk 1 above)
Comment (ram-senth) [reply src/seer/json_api.py]: Have removed use of metrics API and changed the key-value pairs.

File: src/seer/json_api.py (hunk 2 above)
Comment (ram-senth) [reply src/seer/json_api.py]: Used the second option to update existing span.

**Pull request merged**